data/registry_model/registry_proxy_model.py

```diff
@@ -266,6 +277,7 @@ def _create_and_tag_manifest(
         or the retrieved manifest is invalid (for docker manifest schema v1).
         """
         self._proxy.manifest_exists(manifest_ref, ACCEPTED_MEDIA_TYPES)
+        self._enforce_repository_quota(repo_ref)
```

Comment:
any form of force caching should be the final thing to do. If you need a check to ensure the quota isnt exceeded at the beginning do not use force cache for performance and server load reduction.

Reply:
that order of operations would be correct yes. 

Also yes you may be correct that the order of operations in manifest v2 may be incorrect I will re-evaluate that flow.